### PR TITLE
Fix pending runtime api

### LINKED ISSUE #1605: Error deploying contract with Hardhat
I'm trying to deploy a contract but I'm getting an error:

```sh
ProviderError: Create pending runtime api error: Failed to call runtime API, Execution failed: Execution aborted due to trap: wasm trap: wasm `unreachable` instruction executed
WASM backtrace:
error while executing at wasm backtrace:
    0: 0x585386 - <unknown>!rust_begin_unwind
    1: 0x1a646 - <unknown>!core::panicking::panic_fmt::h916104d5f54b6e68
    2: 0x20cfb - <unknown>!core::option::expect_failed::hd86ca5d4fb9d1ec9
    3: 0x484ff8 - <unknown>!<cumulus_pallet_parachain_system::pallet::Pallet<T> as frame_support::inherent::ProvideInherent>::create_inherent::h35a0fb4a0077b731
    4: 0x2d2c48 - <unknown>!<sp_inherents::InherentData as koi_runtime::InherentDataExt>::create_extrinsics::h704a5be8f56289aa
    5: 0x34d2e1 - <unknown>!BlockBuilder_inherent_extrinsics
```
I'm following the [guide](https://docs.darwinia.network/build/ethereum-tools/interact-with-hardhat/#deploy-storage-contract) provided.

It fails on both the **Koi** and **Crab** networks. Note that for Crab, I don't even have any funds in the account. It looks like it fails before the deployment itself.

### PR DESCRIPTION
**Pull request by boundless-forest**
Close #1605 

The current pending RPC is problematic because it lacks the `setValidateData` inherent data provider, which is required by the parachain-system pallet. The pending RPC attempts to [create extrinsics based on the inherent data](https://github.com/darwinia-network/frontier/blob/1fb09fb677efda7db5a9a2cc1391a77b34876876/client/rpc/src/eth/pending.rs#L79 ) and apply them before pool extrinsics, but it fails due to missing inherent data. This PR adds mock data to fulfill the validation requirements.

```sh
2024-10-08 16:01:11.807 DEBUG tokio-runtime-worker eth-pending: Pending runtime API: header digest = Digest { logs: [DigestItem::PreRuntime([97, 117, 114, 97], [204, 123, 43, 17, 0, 0, 0, 0])] }    
2024-10-08 16:01:11.808 ERROR tokio-runtime-worker runtime: panicked at /home/bear/.cargo/git/checkouts/polkadot-sdk-38f315c8db144fd3/5811785/cumulus/pallets/parachain-system/src/lib.rs:933:74:
validation function params are always injected into inherent data; qed  
```

```sh
2024-10-09 16:17:42.006  INFO tokio-runtime-worker aura: 🔖 Pre-sealed block for proposal at 2. Hash now 0x06585aa79fb309bb1d14664f86116a3a879d887089c3f247cb7eb55c5ceac859, previously 0xbc9a01d309b3a9d85b73e000482338b16a792f2470547e80f44fa5a734deb3fc.    
2024-10-09 16:17:42.006  INFO tokio-runtime-worker substrate: ✨ Imported #2 (0x0658…c859)    
2024-10-09 16:17:42.013 DEBUG tokio-runtime-worker bear: Detect the call, number_or_hash: Some(Latest)    
2024-10-09 16:17:42.013 DEBUG tokio-runtime-worker bear: native block is ok, id: BlockId::Hash(0x06585aa79fb309bb1d14664f86116a3a879d887089c3f247cb7eb55c5ceac859)    
2024-10-09 16:17:42.015 DEBUG tokio-runtime-worker bear: Detect the call, number_or_hash: Some(Pending)    
2024-10-09 16:17:42.015 DEBUG tokio-runtime-worker bear: native block is none, ready to create pending runtime api    
2024-10-09 16:17:42.015 DEBUG tokio-runtime-worker eth-pending: Pending runtime API: header digest = Digest { logs: [DigestItem::PreRuntime([97, 117, 114, 97], [177, 180, 43, 17, 0, 0, 0, 0])] }    
2024-10-09 16:17:42.016 DEBUG tokio-runtime-worker eth-pending: Pending runtime API: inherent len = 2    
2024-10-09 16:17:42.016 ERROR tokio-runtime-worker runtime: panicked at /home/bear/.cargo/git/checkouts/polkadot-sdk-38f315c8db144fd3/5811785/cumulus/pallets/parachain-system/src/lib.rs:1297:30:
included head not present in relay storage proof    
2024-10-09 16:17:42.016  WARN tokio-runtime-worker sp_state_machine::overlayed_changes::changeset: 1 storage transactions are left open by the runtime. Those will be rolled back.    
2024-10-09 16:17:42.016  WARN tokio-runtime-worker sp_state_machine::overlayed_changes::changeset: 1 storage transactions are left open by the runtime. Those will be rolled back.    
2024-10-09 16:17:42.016 DEBUG tokio-runtime-worker eth-pending: Pending runtime API: extrinsic len = 0    
2024-10-09 16:17:43.454  INFO tokio-runtime-worker substrate: 💤 Idle (0 peers), best: #2 (0x0658…c859), finalized #2 (0x0658…c859), ⬇ 0 ⬆ 0 
```

image

### CODE REVIEW

File: node/src/service/mod.rs [hunk 1]
```diff
@@ -905,7 +908,19 @@ where
...
+				validation_data: PersistedValidationData {
+					relay_parent_number: u32::MAX,
+					relay_parent_storage_root,
+					..Default::default()
+				},
+				relay_chain_state,
+				downward_messages: Default::default(),
+				horizontal_messages: Default::default(),
+			};
+			Ok((slot, timestamp, parachain_inherent_data))
```
Comment (AurevoirXavier) [inline node/src/service/mod.rs:923]: This part is dev node only.

**Comment by AurevoirXavier** [comment]
Has it been resolved in the dev node?

**Comment by boundless-forest** [comment]
> Has it been resolved in the dev node?

Yes, it has worked well in my local testing.

File: node/src/service/mod.rs [hunk 2]
```diff
@@ -389,16 +390,34 @@ where
...
+					validation_data: cumulus_primitives_core::PersistedValidationData {
+						relay_parent_number: u32::MAX,
+						relay_parent_storage_root,
+						..Default::default()
+					},
+					relay_chain_state,
+					downward_messages: Default::default(),
+					horizontal_messages: Default::default(),
+				};
+			Ok((timestamp, parachain_inherent_data))
```
Comment (AurevoirXavier) [inline node/src/service/mod.rs:420]: https://github.com/moonbeam-foundation/moonbeam/blob/daec7fe4c6ad4f595f8e97e2575716f136a4b240/node/service/src/rpc.rs#L228-L249

Moonbeam's implementation looks simpler.

**Pull request merged**